### ISSUE TITLE
Updating the extension destroys tab groups.

### ISSUE THREAD

**Issue by DaveKap** [gioxx/MarvellousSuspender#76]
I'm on the latest Chrome and Marvelous Suspender just asked if it could update. I saved a session, as suggested, and updated. What I got was 1 tab group staying intact, 1 tab group staying intact but losing about 20 of its tabs, and 2 other tab groups being demolished. The tabs inside are thankfully still up but now I have to reorganize all of them again.

If there's a way to get these groups back with the session saved, I would appreciate the pointer. Otherwise, please fix whatever bug causes updating Marvelous Suspender to improperly work with Chrome's new tab groups.

**Comment by DaveKap** [comment]
I should note that, thankfully, the tabs are in the order they would be if the groups didn't exist and I can shift-select tabs in order to put them all back into the groups they were in, so it isn't really a hassle to get my groups back. Whew!

**Comment by DezG** [comment]
The current version on github is listed  as 7.1.6.1

I wonder what is in this 7.1.6.2 version on the Chrome store!

**Comment by bobsage123** [comment]
After I updated my suspended tabs didn't come back automatically like they used to either. And the backup it makes doesn't show which of my tabs were suspended.

**Comment by sheafitzek** [comment]
I had the same issue with my tab groups disappearing after updating the extension.

**Comment by glade-at-gigwell** [comment]
This needs to get addressed before the next major update.  I've lost a lot of time organizing tabs due to this upgrade, again.

**Comment by civygit** [comment]
Lost tab groups as well. One tab group with one tab remains, the rest are outside.

**Comment by gioxx** [comment]
Hi and thanks for the reports.
We will check this problem with @sherkot to fix TMS behaviour with the tabs new organization on Chrome.

**Comment by Technetium1** [comment]
I'm quite late to this, but you could try to use `CTRL+SHIFT+T` after a restart, if the session is set to continue where you left off it may also allow you to reopen groups.

**Comment by rkodey** [comment]
## In Progress
- ✅ Display tab groups on the Sessions page
- 🔳 Add tab groups to Session export & import
- 🔳 Rebuild tab groups on tab recovery during extension updates

**Comment by rkodey** [comment]
## In Progress
- ✅ Display tab groups on the Sessions page
- ✅ Add tab groups to Session export & import
- ✅ Rebuild tab groups on tab recovery during extension updates
- 🔳 Update session "open" features to handle tab groups

**Comment by rkodey** [comment]
## Resolved
- ✅ Display tab groups on the Sessions page
- ✅ Add tab groups to Session export & import
- ✅ Rebuild tab groups on tab recovery during extension updates
- ✅ Update session "open" features to handle tab groups

Resolved in the current milestone branch, to be included in the next version.

**Comment by diegoteliz** [comment]
I just updated to v8.1.0 and lost all my tab groups again, except for the first one, which was pinned. 

Will this continue to happen on future version upgrades? or will this be the last time? (considering v8.1.0 introduces Tab Groups)

I'm on Vivaldi 7.6.3797.63

**Comment by gioxx** [comment]
@diegoteliz correct. The 8.1.0 introduces just the ability to save (and then rebuild) Tab Groups. That is why you should have "_suffered_" (sorry) the last destruction due to TMS.

**Comment by rkodey** [comment]
@diegoteliz - You should still be able to export one of the saved sessions, and if they were saved with 8.1.0 it should contain all of the Tab Group details.  Restoring the export should work as well...  If the export doesn't contain any personal URLs, would you consider attaching it?  ( Or email via https://github.com/rkodey )

Or, you can look at the export file.  At the end, there will be a `tabGroups` section.  I'm curious if all your tab groups are represented there.

I'll have to run some tests with Vivaldi, if it's handling tab groups differently.
Upgrading from 8.0.0 to 8.1.0 should have restored all your tab groups.

**Comment by diegoteliz** [comment]
@rkodey I didn't save the session when migrating from 8.0.0 to 8.1.0.

My browser (Vivaldi) just notified me that this extension was now disabled. I re-enabled it, and I can see the version is now 8.1.1.

What I can also see is that all my suspended tabs are gone. I've been trying to find them and I can't... and it looks like I'm [not the only one with this problem](https://github.com/gioxx/MarvellousSuspender/issues/314).

I'd love to continue using this extension, but I can't keep losing tabs. It's just too unreliable. Uninstalling...

**Comment by TyIsI** [comment]
I did have multiple sessions saved before my browser (Chrome) updated to v8.1.1, but it broke the sessions and I had to repair, which I did without thinking, and it deleted all suspended tabs.

**Comment by rkodey** [comment]
Closing this issue, in favor of the newer more focused Vivaldi investigation.
